### PR TITLE
feat: implement logical expression

### LINKED ISSUE #23: how about supporting an `or` operation among different expression
Currently multiple expressions in rule params specified in `expr.new` use the `and` semantic in default. All the expressions should be matched at same time. Is there any consideration support the `or` semantic among different expressions?
Here is the example:
```
local ex = expr.new({
        {"user_type", "==", "admin"},
        {"device_type", "in", {"phone"}},
})
```
When we do `eval` method, any ctx table with an "admin" user_type or with a "phone" device_type will return true.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,17 +77,41 @@ Table of Contents
...
+Logical operator can be one of
+* OR
+* AND
+* !OR: not (expr1 or expr2 or ...)
+* !AND: not (expr1 and expr2 and ...)
+
+Their combination can be like:
+
+```json
+{
```
Comment (membphis) [inline README.md]: for `json`, we should use `[` here, it is an array object

File: README.md [hunk 2]
```diff
@@ -54,17 +77,41 @@ Table of Contents
...
+Each expression is an array table which has three or four elements:
 ```lua
 {
     {"var name (aka. left value)", "optional '!' operator", "operator", "const value (aka. right value)"},
     ...
 }
 ```
 
+Logical operator can be one of
+* OR
```
Comment (membphis) [inline README.md:96]: Do we support lowercase expressions, such as'or','and'?

Comment (spacewander) [reply README.md:96]: No. I think we can implement the uppercase only first, as the uppercase is easier to see. Ignore cases may cause the expressions to mix different cases.

File: README.md (hunk 1 above)
Comment (spacewander) [reply README.md]: @membphis 
Fixed.

**Review by membphis — APPROVED** [review]

**Pull request merged**